torch/nn/modules/loss.py

```diff
@@ -108,40 +108,61 @@ class NLLLoss(_WeightedLoss):
     The `input` given through a forward call is expected to contain
     log-probabilities of each class. `input` has to be a Tensor of size either
     :math:`(minibatch, C)` or :math:`(minibatch, C, d_1, d_2, ..., d_K)`
-    with :math:`K \geq 1` for the `K`-dimensional case (described later).
+    with :math:`K \geq 1` for the `K`-dimensional case. The latter is useful for
+    higher dimension inputs, such as computing NLL loss per-pixel for 2D images.
 
     Obtaining log-probabilities in a neural network is easily achieved by
     adding a  `LogSoftmax`  layer in the last layer of your network.
     You may use `CrossEntropyLoss` instead, if you prefer not to add an extra
     layer.
 
-    The `target` that this loss expects should be a class index in the range :math:`[0, C-1]`
-    where `C = number of classes`; if `ignore_index` is specified, this loss also accepts
-    this class index (this index may not necessarily be in the class range).
+    The `target` that this criterion expects should contain either:
+
+    - Class indices in the range :math:`[0, C-1]` where :math:`C` is the number of classes; if
+      `ignore_index` is specified, this loss also accepts this class index (this index
+      may not necessarily be in the class range). The unreduced (i.e. with :attr:`reduction`
+      set to ``'none'``) loss for this case can be described as:
+
+      .. math::
+          \ell(x, y) = L = \{l_1,\dots,l_N\}^\top, \quad
+          l_n = - w_{y_n} x_{n,y_n} \cdot \mathbb{1}\{y_n \not= \text{ignore\_index}\},
+
+      where :math:`x` is the input, :math:`y` is the target, :math:`w` is the weight, and
+      :math:`N` is the batch size. If :attr:`reduction` is not ``'none'``
+      (default ``'mean'``), then
+
+      .. math::
+          \ell(x, y) = \begin{cases}
+              \sum_{n=1}^N \frac{1}{\sum_{n=1}^N w_{y_n} \cdot \mathbb{1}\{y_n \not= \text{ignore\_index}\}} l_n, &
+               \text{if reduction} = \text{`mean';}\\
+                \sum_{n=1}^N l_n,  &
+                \text{if reduction} = \text{`sum'.}
+            \end{cases}
+
+    - Probabilities over classes; useful when labels beyond a single class per minibatch item
```

Comment:
Hm I agree with making it clear that there's one probability per class, but "vector" is a bit confusing to me because the target will be N-dimensional (same as the input shape).

What do you think about just "Probabilities for each class" instead?